src/ring/middleware/json.clj

```diff
@@ -100,12 +102,18 @@
        (handler request respond raise)
        (respond malformed-response)))))
 
+(defn- generate-input-stream
+  [body options]
+  (ring-io/piped-input-stream
+   (fn [out] (json/generate-stream body (io/writer out) options))))
+
 (defn json-response
   "Converts responses with a map or a vector for a body into a JSON response.
   See: wrap-json-response."
-  [response options]
+  [response {:keys [stream?] :as options}]
   (if (coll? (:body response))
-    (let [json-resp (update-in response [:body] json/generate-string options)]
+    (let [generator (if stream? generate-input-stream json/generate-string)
+          json-resp (update-in response [:body] generator (dissoc options :stream?))]
```

Comment:
Please ensure lines are within 80 characters. I have some [contributing guidelines](https://github.com/weavejester/contributing/blob/master/CONTRIBUTING.md) that I've yet to add to all my projects, especially the older ones like Ring-JSON.